mock/api.js

```diff
@@ -21,7 +22,7 @@ export const mockResults = hostname =>
     end: moment.utc(),
     // Since dataset id is a long hex string, removed "-" characters here to make it look like real data
     id: casual.uuid.replace(/-/g, ''),
-    result: casual.word,
+    result: createUniqueKey(),
```

Comment:
can we do something like `casual.word + createUniqueKey` to make it sound a bit like a controller name rather than a hash-like string? I think this will just be for readability purposes

Reply:
I think this is the difficult case; `mockControllers` is a static payload that's generated *once* when `api.js` is loaded, whereas `mockResults` is a function that's run each time there's a call to `/datasets/detail`, meaning that the list of datasets reported for a given controller will change each time we make the call.

Also, we're now using your `createUniqueKey()` for the controller names reported by `/controllers/list` while here we're using `hostname` for the controller. (Line 21.)

That is, we may report a set of controllers for the date range including controller `1234567890abcdef` in `/controllers/list` and then when we call `/datasets/list` with that controller name, we're given a list of datasets with `createUniqueKey` "result" names which claim to have been generated from controller "hostname" instead of `1234567890abcdef`. As I said in the sync meeting, I don't think this impacts the behavior of the summary view page code, but it definitely isn't an accurate "mock" of the real user experience. I worry that this may complicate understanding and testing in the future, even though I can't currently construct any sequence where it's a real problem at present.

An alternative might be to generate a set of `mockResults` for each randomly generated controller name, using that controller name and with a random "result" name, at the same time `mockControllers` is called when we load `api.js`, perhaps in a JSON document using the controller name as key. Then, the "dynamic mock" for `/results/list` would return `mockResults[controller]` and we'd have consistent data across multiple calls. (At the expense of more complicated up-front processing.)

Another possibility would be for the `datasets/list` mock to dynamically generate a `mockResults` payload, but to *cache* the generated value in `mockResults[controller]` and re-use it if called again with the same controller rather than generating a new payload.

Reply:
Similarly, `mockDetail` passes the dataset *name* and uses a randomly generated controller. If these could be connected and cached, the UI "flow" would be more consistent. It probably doesn't matter to the automated tests at this point, but when running with the `start:mock` development server it'd more closely correspond to the real user experience. It's hard to weigh how much this really matters.

Reply:
I think this may also provide another approach to ensuring uniqueness of the keys. You could, for example, generate names in sequence: `controller1`, `controller2` and so forth; and `dataset1`, `dataset2`, and so forth.

Reply:
@dbutenhof to maintain the consistency across the `mockResults` I have used the `hostname`, which is the result of `mockControllers` (so we can safely assume that they are unique), and added a static value(the index of the results) to them for creating result names. In that way every time we call `api.js` with the same controller name, we will get the same result names. 

However, I agree that caching technique  would be more appropriate for the use cases like `overview-page` but I would like to address that in another PR and change the models to accommodate `cache`